pkg/provider/provider.go

```diff
@@ -25,8 +25,11 @@ import (
 // Provider is a common interface for interacting with secret backends.
 type Provider interface {
 	// New constructs a SecretsManager Provider
-	New(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (Provider, error)
+	New(ctx context.Context, store esv1alpha1.GenericStore, kube client.Client, namespace string) (SecretsClient, error)
```

Comment:
Nit: Might be more clear if function is `NewClient(...)`.

I understand that this my proposal, but just re-reading 😄

Reply:
I agree :)